pymove/utils/extract_meets_from_grid.py

```diff
@@ -0,0 +1,252 @@
+    print("-------------------------------------------\n")
+
+    # create a virtual grid to apply in dataframe
+    dic_grid = gridutils.create_virtual_grid(cell_size=cell_size, bbox=bbox)
+    # applying grid in dataframe
+    trajutils.create_update_index_grid_feature(df_, dic_grid=dic_grid)
+
+    # verify if there are locals out of bounding box
+    # if there is not, do nothing and return the dataframe
+    # else, remove such locals
+    out_bbox = df_[(df_['index_grid_lat'] < 0) | (df_['index_grid_lon'] < 0)]['id'].unique()
+
+    if not out_bbox:
+        print("... There is no one cell outside the Grid Boundaries")
+
+    else:
+        print("-----------------------------------------")
+        print("The Following Cells Were Outside The Grid Boundaries")
+        print(out_bbox.tolist())
+        print("-----------------------------------------")
+        df_.drop(df_[df_['id'].isin(out_bbox)].index, inplace=True)
+
+    # df_.reset_index(drop=True, inplace=True)
+    return df_
+
+# ------------------------------------------------------------------------------------------------
+
+def create_distincts_time_slot_in_minute_from_datetime(
+    df,                           
+    slot_interval, 
+    label_time_slot='time_slot',
+    datetime='datetime'
+                                                      ):
+
+    '''
+    Creates a column that represents the unique time_slot referent to datetime
+
+    Parameters
+    ----------
+    df: pandas.core.DataFrame
+        Input trajectory data.
+
+    slot_interval: int
+        Number that represents the time_slot minutes 
+
+    label_time_slot: str, optional, default 'time_slot'
+        name to be given to the time_slot column
+
+    datetime: str, optional, default 'datetime'
+        represents the column name datetime
+
+    '''
+    try:
+        if df['datetime'].dtype == np.dtype('<M8[ns]'):
+
+            day_in_minutes = df.datetime.dt.day * 24 * 60
+            hour_in_minutes = df.datetime.dt.hour * 60
+
+            min_day = day_in_minutes + hour_in_minutes + df.datetime.dt.minute
+
+            df[label_time_slot] = min_day / slot_interval
+            df[label_time_slot] = df[label_time_slot].astype(int)
+            print('time_slot :{}/n', format(min_day / slot_interval))
+
+        else:
+            print('column {} has not a datetime in our dtype')
+
+    except Exception as e:
+        raise e
+
+# ------------------------------------------------------------------------------------------------
+
+def filter_min_max_people(df, min_people, max_people):
+
+    print(df.columns)
+    g = df.groupby(['index_grid', 'time_slot']).agg({'id': 'nunique'}).query(
+        'id >=' + str(min_people) + ' and id <=' + str(max_people))
+    # -----
+    # print("Get N pessoas por time_slot - se falso então não é vazio = {}".format(g.empty))
+    # print(g.head())
+    if g.empty:
+        return "(3º)There is no cell with this amount of people"
```

Comment:
Nesse caso, tem uma função que retorna 2 tipos diferentes, uma `string` ou um `dataframe`.
É melhor criar uma exceção no caso de g for vazio.
Pode fazer assim:
````python
if g.empty:
    raise ValueError("There is no cell with this amount of people")
````